go/ias/api/api.go

```diff
@@ -0,0 +1,40 @@
+// Package api defines the IAS interfaces.
+package api
+
+import (
+	"context"
+
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	"github.com/oasislabs/oasis-core/go/common/sgx/ias"
+)
+
+// Endpoint is an attestation validation endpoint, likely remote.
+type Endpoint interface {
+	// VerifyEvidence takes the provided quote, (optional) PSE manifest, and
+	// (optional) nonce, and returns the corresponding AVR, signature, and
+	// ceritficate chain respectively.
+	VerifyEvidence(ctx context.Context, evidence *Evidence) (*ias.AVRBundle, error)
+
+	// GetSPID returns the SPID and associated info used by the endpoint.
+	GetSPIDInfo(ctx context.Context) (*SPIDInfo, error)
+
+	// GetSigRL returns the Signature Revocation List for a given EPID group.
+	GetSigRL(ctx context.Context, epidGID uint32) ([]byte, error)
+
+	// Cleanup performs post-termination service cleanup.
+	Cleanup()
+}
+
+// SPIDInfo contains information about the SPID associated with the client certificate.
+type SPIDInfo struct {
+	SPID               ias.SPID          `json:"spid"`
+	QuoteSignatureType ias.SignatureType `json:"quote_signature_type"`
+}
+
+// Evidence is attestation evidence.
+type Evidence struct {
```

Comment:
This is not new, it is just moved, previously it was only part of the gRPC protobuf messages.

The RuntimeID is not bound to the quote, it is just used to look up the runtime in the registry to decide whether access should be allowed. The client is free to put in an arbitrary type but the proxy will fetch the given runtime's descriptor and make sure that the quote is for the correct MRENCLAVE.

We could remove this and just use an MRENCLAVE index instead of a RuntimeID to MRENCLAVE index.